File: examples/clj/test/com/rpl/agent/basic/router_agent_test.clj
(ns com.rpl.agent.basic.router-agent-test
  (:require
   [clojure.test :refer [deftest testing is]]
   [com.rpl.agent-o-rama :as aor]
   [com.rpl.rama :as rama]
   [com.rpl.rama.test :as rtest]
   [com.rpl.agent.basic.router-agent :refer [RouterAgentModule]]))

(deftest router-agent-test
  (System/gc)
  (testing "RouterAgent routes messages to appropriate handlers"
    (with-open [ipc (rtest/create-ipc)]
      (rtest/launch-module! ipc RouterAgentModule {:tasks 1 :threads 1})

      (let [manager (aor/agent-manager
                     ipc
                     (rama/get-module-name RouterAgentModule))
            agent   (aor/agent-client manager "RouterAgent")]

        (testing "routes urgent messages to urgent handler"
          (let [result (aor/agent-invoke agent "urgent:system failure")]
            (is (= "[HIGH] system failure" result))))

        (testing "routes non-urgent messages to default handler"
          (let [result (aor/agent-invoke agent "just a message")]
            (is (= "[NORMAL] just a message" result))))))))

File: examples/java/basic/src/test/java/com/rpl/agent/basic/RouterAgentTest.java
package com.rpl.agent.basic;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

import com.rpl.agentorama.AgentClient;
import com.rpl.agentorama.AgentManager;
import com.rpl.rama.test.InProcessCluster;
import com.rpl.rama.test.LaunchConfig;
import org.junit.Test;

/**
 * Test class for RouterAgent demonstrating conditional routing and branching.
 *
 * <p>This test demonstrates:
 *
 * <ul>
 *   <li>Conditional routing based on input values
 *   <li>Different execution paths for different inputs
 *   <li>Router node functionality and branching logic
 * </ul>
 */
public class RouterAgentTest {

  @Test
  public void testRouterAgent() throws Exception {
    // Tests routing for high priority messages
    try (InProcessCluster ipc = InProcessCluster.create()) {
      // Deploy the agent module
      RouterAgent.RouterAgentModule module = new RouterAgent.RouterAgentModule();
      ipc.launchModule(module, new LaunchConfig(1, 1));

      // Get agent manager and client
      String moduleName = module.getModuleName();
      AgentManager manager = AgentManager.create(ipc, moduleName);
      AgentClient agent = manager.getAgentClient("RouterAgent");

      // Test with high priority message
      String result = (String) agent.invoke("urgent:System alert");

      assertNotNull("Result should not be null", result);
      assertTrue("Should contain HIGH priority", result.contains("[HIGH]"));
      assertTrue("Should contain the message", result.contains("System alert"));
    }
  }
}
